salt/auth/mysql.py

```diff
@@ -55,10 +55,39 @@
 log = logging.getLogger(__name__)
 
 try:
+    # Trying to import MySQLdb
     import MySQLdb
-    HAS_MYSQL = True
+    import MySQLdb.cursors
+    import MySQLdb.converters
+    import MySQLdb.constants
+    from MySQLdb.connections import OperationalError
+    HAS_MYSQLDB = True
 except ImportError:
-    HAS_MYSQL = False
+    try:
+        # MySQLdb import failed, try to import PyMySQL
+        import pymysql
+        pymysql.install_as_MySQLdb()
+        import MySQLdb
+        import MySQLdb.cursors
+        import MySQLdb.converters
+        import MySQLdb.constants
+        from MySQLdb.err import OperationalError
+        HAS_PYMYSQL = True
+    except ImportError:
+        HAS_MYSQLDB = HAS_PYMYSQL = False
```

Comment:
I honestly do not like these `UGLY_UPPERCASE_COVARIABLES` all around everywhere, because they are unnecessarily overcomplicating the code so it gets also bulky and hard to read, while value of that is basically zero. I would ask myself: "Do we _really_ care to know what MySQL flavour of driver was loaded? No. We don't really care. So I would not really log that. Therefore:

```python
try:
    import MySQLdb
    ....
except ImportError:
    try:
        import pymysql
        pymysql.install_as_MySQLdb()
        ...
    except ImportError:
        MySQLdb = None
```

At this point you either have a non-`None` object or actually `None`. Then in `__virtual__` function you do:

```python
def __virtual__():
    return bool(MySQLdb), 'No MySQL driver dependency found' if MySQLdb is None else ''
```

Alas, we tend not to log anything in `__virtual__` unless this is really needed. Because you already return the error message as a second element of tuple in `__virtual__` or you just quiet, because it works.